### ISSUE TITLE
Add full report data structure for templates

### ISSUE THREAD

**Issue by carsongee** [aquasecurity/trivy#4072]
I'd like to use additional `types.Report` data in my templates, currently available only in the JSON report. In particular, `ArtifactName`. It seems that `template.go` only renders from `Results`:

https://github.com/aquasecurity/trivy/blob/79a1ba32d5bec7b868dd80ab20bf7e4c4e8eb5a1/pkg/report/template.go#L75

while JSON takes the full data structure:

https://github.com/aquasecurity/trivy/blob/79a1ba32d5bec7b868dd80ab20bf7e4c4e8eb5a1/pkg/report/json.go#L20

Can we change the `template.go` report to include the full `types.Report` data? I'd be happy to do the work myself and refactor the existing `contrib` templates. I know this would have some backwards compatibility issues, so I understand if we'd prefer to keep it as it currently is. Thanks!

**Comment by knqyf263** [comment]
Thanks for your suggestion. Yes, it brings breaking change and will be a big impact. We were not sure how many users needed metadata in templates, and wondering if it was worth causing confusion. Can we hold it for a while and see the demand from the community?

**Comment by carsongee** [comment]
Yep! I understand, and I'm comfortable with that approach. Our primary use case is collating multiple scans of artifact bundles when scanning Helm charts for example. We can continue to do that with external tooling.

Additionally, the current HTML template uses an awkward 0-index into Targets to title the report that would likely make more sense as the Artifact Name: https://github.com/aquasecurity/trivy/blob/914c6f0921c2811c9a62aad9ae2b2dbd1ae21a0d/contrib/html.tpl#L85.  That can be fixed moving the Target into the results and having a generic report title though. 

As I see it, nothing critical at this point requires it, but I see it getting harder and harder to make this type of change as the project matures if it is ever desired. Regardless, thanks for your consideration and the great project!
